### DISCUSSION TITLE
encryption with age, multiple computers

### DISCUSSION THREAD

**Discussion by xvandish** [twpayne/chezmoi#1538] [Q&A]
Hi,

First, thanks for all the work you've put into `chezmoi`!

I've got computer A and computer B - the problem i'm encountering is initializing the encrypted set of config files from A onto B.

I started using chezmoi on A a while back, and used the `age` encryption.
My `~/.config/chezmoi/chezmoi.toml` looked like this at first. 

```
encryption = "age"
[age]
    identity = "/Users/rodrigosilvamendoza/age-key.txt"
    recipient = "computerAPublicKey"
```

This worked great on a single computer.

Then I went to use chezmoi on computer B, and the configs are as follows:
A 
```
encryption = "age"
[age]
    identity = "/Users/rodrigosilvamendoza/age-key.txt"
    recipients = ["computerAPublicKey", "computerBPublicKey"]
```

B
```
encryption = "age"
[age]
    identity = "~/age-key.txt"
    recipients = ["computerAPublicKey", "computerBPublicKey"]
```

After adding the second recipient to my config on computer A, I ran `chezmoi add --encrypt [file]` for every file that `chezmoi` manages, and committed the results

On computer B the following happens  
```
$chezmoi init chezmoi init https://github.com/xvandish/dotfiles.git
$chezmoi status
age: error: no identity matched any of the recipients
age: report unexpected or unhelpful errors at https://.....
chezmoi: exit status 1
```
As an age test, I encrypted a random file on computer A like so
`age -o example.rc.age -r computerAPublicKey -r computerBPublicKey ~/.config.starship.toml`

Then I sent that encrypted file to computer B, and ran 
`age --decrypt -i ~/age-key.txt -o rcfile.toml ~/Downloads/example.rc.age`

and was able to view the contents of `rcfile.toml` just fine.

Any ideas?

**Comment by twpayne** [comment]
Thanks for your kind words and thanks for reporting this.

What's the output of `chezmoi doctor`?

There were some [fixes around age support](https://github.com/twpayne/chezmoi/commit/d2bac3a586aebaaa5bb3a5adeba27fee11aefc4a) that got added in chezmoi v2.7.0, but there's no current test in chezmoi for this, so chezmoi could well be broken here.

**Reply by xvandish** [reply]
Sorry for the delayed reply, I never got a notification from GitHub! And very welcome, this an incredible project with a lot of work put into it. I'll check in later today with the output of `chezmoi doctor ` when I have a chance to check.
